test/core/security/local_security_connector_test.cc

```diff
@@ -0,0 +1,165 @@
+//
+//
+// Copyright 2024 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include "src/core/lib/security/security_connector/tls/tls_security_connector.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "absl/log/check.h"
+
+#include <grpc/credentials.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
+
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/config/config_vars.h"
+#include "src/core/lib/gprpp/crash.h"
+#include "src/core/lib/gprpp/unique_type_name.h"
+#include "src/core/lib/security/context/security_context.h"
+#include "src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.h"
+#include "src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h"
+#include "src/core/lib/security/credentials/tls/tls_credentials.h"
+#include "src/core/tsi/transport_security.h"
+#include "test/core/test_util/test_config.h"
+#include "test/core/test_util/tls_utils.h"
+#include "googletest/include/gtest/gtest.h"
+
+namespace grpc_core {
+namespace testing {
+
+class LocalSecurityConnectorTest : public ::testing::Test {
```

Comment:
This is not needed. You can remove this and use `TEST` instead of `TEST_F` below.

Reply:
I saw 2 styles in surrounding tests - with test related class and without. Which one is preferred? I liked the idea of keeping PrintAuthContext debugging function under this class, but dropped it because wasn't sure about the performance.

Reply:
Having this test class (rather than using the autogenerated one) is unnecessary unless you have setup that you need to do (which we don't in this case). If you see this in other files, I'm guessing what happened is: there used to be setup, the setup got deleted, and they didn't bother to also delete the test class.